### PR TITLE
Re-export config-plugins from expo package and update docs

### PR DESCRIPTION
**Pull request by brentvatne**
# Why

https://github.com/jakobo/expo-community-flipper/issues/16

Fixes ENG-6059

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md).

### CODE REVIEW

File: docs/pages/guides/config-plugins.md [hunk 1]
```diff
@@ -498,7 +498,53 @@ Because of this reasoning, the root of a Node module is searched instead of righ
...
+  "peerDependenciesMeta": {
+    "expo": {
+      "optional": true
+    }
+  }
+}
+```
+
+- You may update the exact versions of `@expo/config-types` and `expo` to build against a specific version.
+- For simple config plugins that depend on core, stable APIs, such as a plugin that only modifies **Info.plist** or **AndroidManifest.xml**, you can use a loose dependency such as in the example above.
```
Comment (amandeepmittal) [inline docs/pages/guides/config-plugins.md:524]:
```suggestion
- For simple config plugins that depend on core, stable APIs, such as a plugin that only modifies **Info.plist** or **AndroidManifest.xml**, you can use a loose dependency as shown in the example above.
```

**Review by amandeepmittal — APPROVED** [review]
LGTM 👍

**Review by tsapeta — APPROVED** [review]

**Review by EvanBacon — APPROVED** [review]

**Comment by brentvatne** [comment]
The following packages currently use private APIs from `@expo/config-plugins`, so I didn't change them to use the new format yet:

- expo-build-properties: `@expo/config-plugins/build/utils/generateCode`, `@expo/config-plugins/build/plugins/android-plugins` (for mocking in test), `@expo/config-plugins/build/plugins/ios-plugins` (for mocking in test)
- expo-camera: `@expo/config-plugins/build/utils/generateCode`
- expo-image-picker: `@expo/config-plugins/build/android/Manifest`
- expo-dev-client:  `@expo/config-plugins/build/android/Manifest`

I think we should fix these in a follow-up PR. @EvanBacon - should we be exposing `Manifest` and `generateCode` through the public API?

**Comment by KiwiKilian** [comment]
[@rnmapbox/maps](https://github.com/rnmapbox/maps) also uses some private APIs which are currently not exported with the new style:

```js
import {
  mergeContents,
  removeGeneratedContents,
} from '@expo/config-plugins/build/utils/generateCode';
```

Do you already have any followup issue on this topic?

**Comment by brentvatne** [comment]
@KiwiKilian - you should be able to copy and paste those functions out into the mapbox library, these aren't intended to be public APIs. ideally we want to come up with a better API for interacting with podfile

**Comment by SimenB** [comment]
Official plugins also use it. https://github.com/expo/config-plugins/blob/ed020d236b8fa4a810adf90a9773d9a2fba36fd4/packages/react-native-pdf/src/withPdf.ts#L7

Maybe expose it as `unstable` or something up until a replacement is made?

**Pull request merged**